File: src/main/java/io/github/komdosh/alarms/actions/AlarmDisable.java
package io.github.komdosh.alarms.actions;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.components.ServiceManager;
import io.github.komdosh.alarms.services.AlarmService;

public class AlarmDisable extends AnAction {

    public AlarmDisable() {
        super("AlarmDisable");
    }

    public void actionPerformed(AnActionEvent event) {
        AlarmService alarmService = ServiceManager.getService(AlarmService.class);
        alarmService.disableAlerts();
    }
}

File: src/main/kotlin/io/github/komdosh/alarms/actions/AlarmDisable.kt
package io.github.komdosh.alarms.actions

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.components.ServiceManager
import io.github.komdosh.alarms.services.AlarmService

class AlarmDisable : AnAction("AlarmDisable") {
    override fun actionPerformed(event: AnActionEvent) {
        event.project?.let{
            val alarmService = ServiceManager.getService(AlarmService::class.java)
            alarmService.disableAlerts(it)
        }
    }
}
